perun/postprocess/regressogram/methods.py

```diff
@@ -23,59 +30,64 @@ def compute(data_gen, configuration):
     :return: list of dict: the computation results
     """
     # checking the presence of specific keys in individual methods
-    validate_dictionary_keys(configuration, _REQUIRED_KEYS, [])
+    tools.validate_dictionary_keys(configuration, _REQUIRED_KEYS, [])
 
     # list of result of the analysis
     analysis = []
-    for chunk in data_gen:
-        result = regressogram(chunk[0], chunk[1], configuration['statistic'], configuration['bins'])
-        result['uid'] = chunk[2]
+    for x_pts, y_pts, uid in data_gen:
+        # Check whether the user gives as own number of buckets or select the method to its estimate
+        buckets = configuration['bucket_number'] if configuration.get('bucket_number') else configuration[
```

Comment:
Suggestion: If the conditional is checking whether 'bucket_number' is in the configuration, then I would suggestion to simplify this to `buckets = configuration.get('bucket_number', configuration['bucket_method'])`, i.e. if 'bucket_number' is in configuration used it, otherwise as default use the computed one (though I guess it could call the underlying method)

Reply:
It is a good idea, but _click_ configuration also includes the keys of parameters, which were not entered, with default value **None**. That's why the condition `configuration.get('bucket_number')` is valid due to the described behavior.